Firebase/DynamicLinks/FDLURLComponents/FDLURLComponents.m

```diff
@@ -448,15 +449,51 @@ - (instancetype)init {
 
 @implementation FIRDynamicLinkComponents
 
+#pragma mark Deprecated Initializers.
 + (instancetype)componentsWithLink:(NSURL *)link domain:(NSString *)domain {
   return [[self alloc] initWithLink:link domain:domain];
 }
 
 - (instancetype)initWithLink:(NSURL *)link domain:(NSString *)domain {
+  NSURL *domainURL = [NSURL URLWithString:domain];
+  if (!domainURL.scheme) {
+    FDLLog(FDLLogLevelWarning, FDLLogIdentifierSetupWarnHTTPSScheme,
+           @"Only https scheme is allowed. The supplied domain's scheme will be treated as https.");
+  }
+  NSString *domainURIPrefix =
+      domainURL.scheme ? domain : [NSString stringWithFormat:@"https://%@", domain];
+  self = [super init];
+  if (self) {
+    _link = link;
+    _domain = domainURIPrefix;
+  }
+  return self;
+}
+
+#pragma mark Initializers.
++ (instancetype)componentsWithLink:(NSURL *)link domainURIPrefix:(NSString *)domainURIPrefix {
+  return [[self alloc] initWithLink:link domainURIPrefix:domainURIPrefix];
+}
+
+/// Deprecated init API has nonnull return type. If the check fails, support returning an instance
```

Comment:
I think this documentation is now out of date.

Reply:
Done.